Gordon360/ApiControllers/DirectMessageController.cs

```diff
@@ -218,11 +215,11 @@ public IHttpActionResult StoreUserConnectionIds([FromBody] String connectionId)
             /// <returns>true if successful</returns>
             [HttpPut]
             [Route("userConnectionIds")]
-            public IHttpActionResult GetConnectionIds([FromBody] String userId)
+            public IHttpActionResult GetConnectionIds([FromBody] List<string> userIds)
```

Comment:
The structure of each of our rooms have all the ids of the users in that room, we use the Id system to keep track of each unique users, but along with that, when we send a message to a room we also use the ids of all the users in that room to grab their saved connection ids in order to send the message that was sent in the room in real-time to all those users through our WebSocket. And you are right about the PUT, It was the most familiar way for us to send that list. I could be mistaken but this other way seems as though we need to pass in an already specified-sized list. Is there a way to pass in a list object through a GET? other wise it would compromise the way we set up our system of sending information.

Reply:
Firstly, regarding passing the connection ids as query parameters so you can use GET rather than PUT, I don't believe there's any reason you would need to know the length of the list ahead of time. Here's a simple JavaScript excerpt that would turn an array of userIds into a string having those as query parameters:
```
const queryParams = '?userIds=' + userIds.join('&userIds=');
```
For example, if your userIds were the array `['1', '99', '999999099']`, then you would get `queryParams` = `?userIds=1&userIds=99&userIds=999999099`.

On the C# route, you would have `GetConnectionIds([FromUri] List<String> userIds)` and `userIds` would automatically be populated as the List `["1", "99", "999999099"]`.

Reply:
Secondly, I still don't fully understand what the use case for this route is. Are you saying that when a user sends a message to a room, you get all the userIds associated with that room and then use those userIds to fetch all the connectionIds associated with that room, and then send the message to all those connectionIds? It seems to me like the front end could just send a message to a route like `SendMessageToGroup(String groupID)`, and then the backend would take care of getting all the connectionIds associated with that groupID and send the message to them. Why does the front end need to know the userIds or connectionIds or the users associated with a group? As long as the front end knows which group it's sending to, shouldn't the backend be able to take care of the rest?

I obviously don't know the structure of your app or the conventions of SignalR as well as you, so please correct me if I'm missing something. I just want to make sure this isn't more complicated than it needs to be.